contracts/minter-imago/src/contract.rs

```diff
@@ -0,0 +1,743 @@
+            .querier
+            .query_wasm_smart(wl, &WhitelistQueryMsg::Config {})?;
+
+        if res.is_active {
+            return Err(ContractError::WhitelistAlreadyStarted {});
+        }
+    }
+
+    config.whitelist = Some(deps.api.addr_validate(whitelist)?);
+    CONFIG.save(deps.storage, &config)?;
+
+    Ok(Response::default()
+        .add_attribute("action", "set_whitelist")
+        .add_attribute("whitelist", whitelist.to_string()))
+}
+
+pub fn execute_mint_sender(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+) -> Result<Response, ContractError> {
+    let config = CONFIG.load(deps.storage)?;
+    let action = "mint_sender";
+
+    // If there is no active whitelist right now, check public mint
+    // Check if after start_time
+    if is_public_mint(deps.as_ref(), &info)? && (env.block.time < config.start_time) {
+        return Err(ContractError::BeforeMintStartTime {});
+    }
+
+    // Check if already minted max per address limit
+    let mint_count = mint_count(deps.as_ref(), &info)?;
+    if mint_count >= config.per_address_limit {
+        return Err(ContractError::MaxPerAddressLimitExceeded {});
+    }
+
+    _execute_mint(deps, env, info, action, false, None, None)
+}
+
+// Check if a whitelist exists and not ended
+// Sender has to be whitelisted to mint
+fn is_public_mint(deps: Deps, info: &MessageInfo) -> Result<bool, ContractError> {
+    let config = CONFIG.load(deps.storage)?;
+
+    // If there is no whitelist, there's only a public mint
+    if config.whitelist.is_none() {
+        return Ok(true);
+    }
+
+    let whitelist = config.whitelist.unwrap();
+
+    let wl_config: WhitelistConfigResponse = deps
+        .querier
+        .query_wasm_smart(whitelist.clone(), &WhitelistQueryMsg::Config {})?;
+
+    if !wl_config.is_active {
+        return Ok(true);
+    }
+
+    let res: HasMemberResponse = deps.querier.query_wasm_smart(
+        whitelist,
+        &WhitelistQueryMsg::HasMember {
+            member: info.sender.to_string(),
+        },
+    )?;
+    if !res.has_member {
+        return Err(ContractError::NotWhitelisted {
+            addr: info.sender.to_string(),
+        });
+    }
+
+    // Check wl per address limit
+    let mint_count = mint_count(deps, info)?;
+    if mint_count >= wl_config.per_address_limit {
+        return Err(ContractError::MaxPerAddressLimitExceeded {});
+    }
+
+    Ok(false)
+}
+
+pub fn execute_mint_to(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    recipient: String,
+) -> Result<Response, ContractError> {
+    let recipient = deps.api.addr_validate(&recipient)?;
+    let config = CONFIG.load(deps.storage)?;
+    let action = "mint_to";
+
+    // Check only admin
+    if info.sender != config.admin {
+        return Err(ContractError::Unauthorized(
+            "Sender is not an admin".to_owned(),
+        ));
+    }
+
+    _execute_mint(deps, env, info, action, true, Some(recipient), None)
+}
```

Comment:
maybe imago should not support airdrop?

Reply:
It's not necessary, but I wanted to keep it in order for the creator to gift nfts to others.